Tests/Ecommerce/TestEcommerceEvents.swift

```diff
@@ -0,0 +1,282 @@
+        XCTAssertEqual("id", entity.data["transaction_id"] as? String)
+        XCTAssertEqual(55.55, entity.data["revenue"] as? Double)
+        XCTAssertEqual("CAD", entity.data["currency"] as? String)
+        XCTAssertEqual("cash", entity.data["payment_method"] as? String)
+        XCTAssertEqual(2, entity.data["total_quantity"] as? Int)
+        XCTAssertEqual(11.11, entity.data["tax"] as? Double)
+        XCTAssertEqual(0, entity.data["shipping"] as? Double)
+        XCTAssertEqual("new", entity.data["discount_code"] as? String)
+        XCTAssertEqual(0.5, entity.data["discount_amount"] as? Double)
+        XCTAssertEqual(false, entity.data["credit_order"] as? Bool)
+    }
+    
+    func testRefund() {
+        let event = RefundEvent(
+            transactionId: "id",
+            refundAmount: 300,
+            currency: "INR",
+            refundReason: "reason",
+            products: [product1]
+        )
+        let entities = event.entitiesForProcessing
+        let refundEntities = getRefundEntities(entities)
+        
+        XCTAssertEqual(ecommerceActionSchema, event.schema)
+        XCTAssertEqual("refund", event.payload["type"] as? String)
+        XCTAssertEqual(2, entities?.count)
+        XCTAssertEqual(1, refundEntities.count)
+        XCTAssertEqual(1, getProductEntities(entities).count)
+        
+        let entity = refundEntities[0]
+
+        XCTAssertEqual("id", entity.data["transaction_id"] as? String)
+        XCTAssertEqual("INR", entity.data["currency"] as? String)
+        XCTAssertEqual(300, entity.data["refund_amount"] as? Double)
+        XCTAssertEqual("reason", entity.data["refund_reason"] as? String)
+    }
+    
+    func testPromotionClick() {
+        let event = PromotionClickEvent(promotion: PromotionEntity(id: "promo"))
+        let entities = event.entitiesForProcessing
+        
+        XCTAssertEqual(ecommerceActionSchema, event.schema)
+        XCTAssertEqual("promo_click", event.payload["type"] as? String)
+        XCTAssertEqual(1, entities?.count)
+        XCTAssertEqual(1, getPromotionEntities(entities).count)
+    }
+    
+    func testPromotionView() {
+        let event = PromotionViewEvent(promotion: PromotionEntity(id: "promo"))
+        let entities = event.entitiesForProcessing
+        
+        XCTAssertEqual(ecommerceActionSchema, event.schema)
+        XCTAssertEqual("promo_view", event.payload["type"] as? String)
+        XCTAssertEqual(1, entities?.count)
+        XCTAssertEqual(1, getPromotionEntities(entities).count)
+    }
+    
+    private let product1 = ProductEntity(
+        id: "id1",
+        category: "category1",
+        currency: "GBP",
+        price: 1.23
+    )
+    
+    let product2 = ProductEntity(
+        id: "id2",
+        category: "category2",
+        currency: "GBP",
+        price: 0.99
+    )
+    
+    private func getProductEntities(_ all: [SelfDescribingJson]?) -> [SelfDescribingJson] {
+        var entities: [SelfDescribingJson] = []
```

Comment:
Nit: this is completely fine, just a note for the future that you can do this with one line:

```
return all?.filter { $0.schema == ecommerceProductSchema }
```